### PR TITLE
change: [M3-8199] - Auto-populate Image Label based on linode and disk names

### PR DESCRIPTION
**Pull request by hkhalil-akamai**
## Description 📝
Auto-populate the Image Label field based on the selected Linode and Disk.

## Changes  🔄
- Auto-populate image labels using the following format: `linode_label-disk_label`.
- Change field label from "Label" to "Image Label" based on UX mockups
- Add unit tests

## Target release date 🗓️
7/8

## Preview 📷
| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/linode/manager/assets/122488130/f0c67060-a497-486f-8002-266adb3a45c4" /> | <video src="https://github.com/linode/manager/assets/122488130/20f200fe-b8ce-41f0-afe9-5071488062c2" /> |

### Prerequisites
- Go to `images/create/disk`

### Reproduction steps
- Select a Linode and Disk and observe the Image Label field is not auto-populated

### Verification steps
- Select a Linode and Disk and observe the Image Label field is auto-populated in the following format: `linode_label-disk_label`.

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/Images/ImagesCreate/CreateImageTab.tsx [hunk 1]
```diff
@@ -89,6 +90,16 @@ export const CreateImageTab = () => {
     }
   });
 
+  const { data: linodes } = useAllLinodesQuery();
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Images/ImagesCreate/CreateImageTab.tsx:93]: Can we use the existing `useLinodeQuery` instead of relying on `useAllLinodesQuery`?

Even though it might take years at this rate, I still want to phase out `useAllLinodesQuery` as much as possible

File: packages/manager/src/features/Images/ImagesCreate/CreateImageTab.tsx [hunk 2]
```diff
@@ -256,10 +273,10 @@ export const CreateImageTab = () => {
                   disabled={isImageCreateRestricted}
                   errorText={fieldState.error?.message}
                   inputRef={field.ref}
-                  label="Label"
+                  label="Image Label"
                   noMarginTop
                   onBlur={field.onBlur}
-                  value={field.value ?? ''}
+                  value={field.value}
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Images/ImagesCreate/CreateImageTab.tsx:279]: This change causes a `A component is changing an uncontrolled input to be controlled` error.
We should either revert this change or set label to an empty string in react hook from's default values.


https://github.com/linode/manager/assets/115251059/c059c948-a5cd-47e6-906d-13ff1262edbe

File: packages/manager/src/features/Images/ImagesCreate/CreateImageTab.tsx [hunk 3]
```diff
@@ -256,10 +273,10 @@ export const CreateImageTab = () => {
                   disabled={isImageCreateRestricted}
                   errorText={fieldState.error?.message}
                   inputRef={field.ref}
-                  label="Label"
+                  label="Image Label"
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Images/ImagesCreate/CreateImageTab.tsx:276]: I assume this change came from the UX mockups. I didn't make this change in my initial refactor of this page because I don't agree with the change 😅 

I find it repetitive to say `Image Label` when we are on the _Image Create_ page while in a paper with the title _Image Details_, but that's just my opinion 🤷‍♂️

**Review by bnussman-akamai — COMMENTED** [review]
Should we clear the label field if the user clears the Linode / Disk?


https://github.com/linode/manager/assets/115251059/591a8e9a-14d6-4cde-aebe-45101c0eb7d1

File: packages/manager/src/features/Images/ImagesCreate/CreateImageTab.tsx (hunk 3 above)
Comment (hkhalil-akamai) [reply packages/manager/src/features/Images/ImagesCreate/CreateImageTab.tsx:276]: Agreed. Let me reach out to UX with the suggestion.

**Review by bnussman-akamai — APPROVED** [review]

File: packages/manager/src/features/Images/ImagesCreate/CreateImageTab.tsx (hunk 3 above)
Comment (hkhalil-akamai) [reply packages/manager/src/features/Images/ImagesCreate/CreateImageTab.tsx:278]: Reverting to "Label"
